src/com/nutrons/steamworks/Shooter.java

```diff
@@ -41,29 +43,46 @@ public Shooter(LoopSpeedController shooterController, Flowable<Boolean> shooterB
     this.shooterButton = shooterButton;
     this.distance = distance;
     this.setpointHint = setpointHint;
-    this.variableSetpoint = this.distance.filter(x -> x != 0.0).map(x -> 11.778 * x + 1858.7).share();
+    this.variableSetpoint = this.distance.filter(x -> x != 0.0).map(x -> 11.778 * x + 1808.7).share();
+  }
+
+  public Command auto() {
+    Flowable<ControllerEvent> setpoint = Flowable.just(AUTO_SETPOINT).map(aimEvent);
+    return Command.fromSubscription(() ->
+        setpoint.subscribe(shooterController))
+        .addFinalTerminator(() -> shooterController.accept(stopEvent));
   }
 
   public Command pulse() {
     Flowable<ControllerEvent> combined = setpointHint.withLatestFrom(Flowable.just(SETPOINT)
-        .mergeWith(variableSetpoint.take(1)), (x, y) -> x + y).map(aimEvent);
+        .mergeWith(
+            toFlow(() ->
+                //this.prefs.getDouble("setpoint_cat", 3000)
+                3000.0
+            )//variableSetpoint.take(1)
+        ), (x, y) -> x + y).map(aimEvent);
     return Command.fromSubscription(() ->
         combined.subscribe(shooterController))
         .addFinalTerminator(() -> shooterController.accept(stopEvent));
   }
 
   @Override
   public void registerSubscriptions() {
+    //this.prefs = Preferences.getInstance();
     this.shooterController.setControlMode(ControlMode.MANUAL);
     this.shooterController.setReversedSensor(true);
     this.shooterController.setPID(PVAL, IVAL, DVAL, FVAL);
-    Consumer<Double> speed = new WpiSmartDashboard().getTextFieldDouble("shooter speed");
+    /**Consumer<Double> speed = new WpiSmartDashboard().getTextFieldDouble("shooter speed");
     toFlow(this.shooterController::speed).subscribe(speed);
-    this.variableSetpoint.subscribe(new WpiSmartDashboard().getTextFieldDouble("calculated setpoint"));
+    this.variableSetpoint.subscribe(new WpiSmartDashboard().getTextFieldDouble("calculated setpoint"));**/
     this.shooterButton.filter(x -> x).map(x -> pulse().terminable(shooterButton.filter(y -> !y)))
         .subscribe(x -> x.execute(true));
 
-    toFlow(this.shooterController::speed).withLatestFrom(this.variableSetpoint, (x, y) -> x + 100 > y && x - 100 < y).onBackpressureDrop()
-        .subscribe(new WpiSmartDashboard().getTextFieldBoolean("shooter rpm within range GO!!"));
+    toFlow(this.shooterController::speed).withLatestFrom(this.variableSetpoint, (x, y) -> x + 100 > y && x - 100 < y)
+        .onBackpressureDrop().map(x -> RobotBootstrapper.feeder.pulse().terminable(shooterButton.filter(y -> !y)))
+        .subscribe(x -> x.execute(true));
```

Comment:
Lol, so whenever we are in setpoint range we will start shooting, instead of having to press the button? I mean I guess it works XD. I would instead put this in robot bootstrapper, and add this one giant toFlow, make it a Command.fromSwitch, and register it in tele. Look at the command.fromSwitch documentation. Then, we won't need a sketchy public static feeder.